packages/lab/src/number-input/NumberInput.tsx

```diff
@@ -202,40 +217,95 @@ function getNumberPrecision(num: number | string) {
 const defaultPattern: NumberInputProps["pattern"] = (inputValue) =>
   /^[+-]?(\d+(\.\d*)?|\.\d*)?$/.test(inputValue);
 
+const defaultDecrement = (
+  value: string,
+  step: number,
+  stepMultiplier: number,
+  parse: (v: string) => number | null,
+) => {
+  const parsedValue = parse(value) ?? 0;
+  const decrementStep = step * stepMultiplier;
+  return String(parsedValue - decrementStep);
+};
+
+const defaultIncrement = (
+  value: string,
+  step: number,
+  stepMultiplier: number,
+  parse: (v: string) => number | null,
+) => {
+  const parsedValue = parse(value) ?? 0;
+  const incrementStep = step * stepMultiplier;
+  return String(parsedValue + incrementStep);
+};
+
+const defaultFormat = (value: string, decimalScale: number): string => {
+  const sanitized = value.trim();
+  if (!sanitized.length) {
+    return "";
+  }
+  const floatValue = Number.parseFloat(sanitized);
+  const updatedValue = Number.isNaN(floatValue)
+    ? sanitized
+    : floatValue.toFixed(decimalScale);
+  return String(updatedValue);
+};
+
+const defaultParse = (value: string, decimalScale: number): number | null => {
+  const sanitizedValue = value.trim();
+  if (!sanitizedValue.length) {
+    return null;
+  }
+  if (
+    sanitizedValue === "." ||
+    sanitizedValue === "+" ||
+    sanitizedValue === "-"
+  ) {
+    return 0;
+  }
+  const floatString = Number.parseFloat(value).toFixed(decimalScale);
```

Comment:
Should this be `value` or `sanitizedValue` ?

Reply:
sanitizedValue is used to return early and only offers a trimmed value, which  parseFloat does not need.